From 052185f74ea7c170bfdd8a8e02300cd94b461a45 Mon Sep 17 00:00:00 2001
From: Jesse Becker <hawson@gmail.com>
Date: Fri, 26 Sep 2025 22:18:20 -0400
Subject: [PATCH] use long-format CLI option for --no-browser to work wth
 syncthing 2.0+

---
 syncthing_gtk/app.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/syncthing_gtk/app.py b/syncthing_gtk/app.py
index dd744a8..f57e4e7 100644
--- a/syncthing_gtk/app.py
+++ b/syncthing_gtk/app.py
@@ -501,7 +501,7 @@ def ct_process(self):
         Sets self.process, adds related handlers and starts daemon.
         Just so I don't have to write same code all over the place.
         """
-        cmdline = [self.config["syncthing_binary"], "-no-browser"]
+        cmdline = [self.config["syncthing_binary"], "--no-browser"]
         vars, preargs, args = parse_config_arguments(self.config["syncthing_arguments"])
         cmdline = preargs + cmdline + args
         if self.home_dir_override:
